#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros_utils'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()
gen.add("connect_angular_threshold", double_t, 0,
        "angular threshold to connect planes", 0.1, 0.0, 1.57)
gen.add("connect_distance_threshold", double_t, 0,
        "distance threshold to connect planes", 0.1, 0.0, 10.0)
gen.add("connect_perpendicular_distance_threshold", double_t, 0,
        "perpendicular distance threshold to connect planes", 0.01, 0.0, 10.0)
gen.add("ransac_refinement_max_iteration", int_t, 0,
        "maximum number of iteration of ransac refinement", 100, 0, 1000)
gen.add("ransac_refinement_outlier_threshold", double_t, 0,
        "outlier threshold of ransac refinement", 0.1, 0.0, 1.0)
# gen.add("ransac_refinement_eps_distance", double_t, 0,
#         "ransac refinement of eps distance", 0.1, 0.0, 1.0)
gen.add("ransac_refinement_eps_angle", double_t, 0,
        "ransac refinement of eps angle", 0.1, 0.0, 1.0)
gen.add("min_size", int_t, 0, "minimum indices size", 100, 0, 1000)
gen.add("max_area", double_t, 0, "the max area of the convex areas", 100, 0, 100)
gen.add("min_area", double_t, 0, "the minimum area of the convex areas", 0.1, 0, 5)
exit (gen.generate (PACKAGE, "jsk_pcl_ros_utils", "PlaneConcatenator"))
